include/circt/Dialect/FIRRTL/FIRRTLUtils.h

```diff
@@ -168,7 +169,14 @@ inline FIRRTLType mapBaseType(FIRRTLType type,
 /// Given a type, return the corresponding lowered type for the HW dialect.
 /// Non-FIRRTL types are simply passed through. This returns a null type if it
 /// cannot be lowered.
-Type lowerType(Type type);
+/// If the given type is a named bundle, then create a TypeScope at the Module
+/// level and the corresponding typedecl op inside it and return the
+/// corresponding TypeAliasType.
+Type lowerType(
+    Type type, std::optional<Location> loc = {},
+    std::optional<
+        llvm::function_ref<hw::TypeAliasType(Type, BundleType, Location)>>
+        getTypeDeclFn = {});
```

Comment:
For nullable function pointer, I think we are using `std::function<>` in other places.